lua/ide/workspaces/workspace_controller.lua

```diff
@@ -246,25 +246,42 @@ function WorkspaceController.new(config)
         -- issues, so do it only after "VimEnter" fires.
         local restore = libwin.restore_cur_win()
         local init_autocmd = nil
-        init_autocmd =  vim.api.nvim_create_autocmd({ "VimEnter" }, {
-            callback = function()
-                -- do not assign if we are git tool.
-                local buf_name = vim.api.nvim_buf_get_name(0)
-                if vim.fn.match(buf_name, ".git/*") >= 0 then
+        local function init_callback()
+            -- do not assign if we are git tool.
+            local buf_name = vim.api.nvim_buf_get_name(0)
+            if vim.fn.match(buf_name, "%.git/") then
```

Comment:
Why is the match string changed here? Im not familiar with that syntax. Any reason not to keep the older one which seems to work as of today

Reply:
`.` means "any character", in Lua pattern matching `%` is the escape character, so `%.` means a literal `.` character. Without this change, `nvim-ide` doesn't launch for me on any project because my base path is `~/git/[github or gitlab]/[repo name]/` so it matches `/git/` as well as `/.git/` when it should only match `/.git/`

Reply:
Oh wait, this is not a Lua match anymore since it's using `vim.fn.match`, I will fix it

Reply:
ah, we need to escape with `\\.`

Reply:
Done.

Reply:
The escape is not necessary. 

Do this test:

```
: lua print(vim.inspect(vim.fn.match('.git/xyz', '.git/*')))
```
prints 0


```
: lua print(vim.inspect(vim.fn.match('.gi/xyz', '.git/*')))
```
prints -1

Reply:
Not sure why it does not work for you?

Reply:
fixed, sorry about that